crates/burn-train/src/metric/confusion_matrix.rs

```diff
@@ -0,0 +1,161 @@
+use super::{ClassificationAverage, ClassificationInput};
+use burn_core::prelude::{Backend, Int, Tensor};
+use std::fmt::{self, Debug};
+
+#[derive(Clone)]
+pub struct ConfusionMatrix<B: Backend> {
+    confusion_matrix_map: Tensor<B, 2, Int>,
+    classification_average: ClassificationAverage,
+}
```

Comment:
This is not actually a confusion matrix?

Looking at the implementation, it looks like an interface to compute the true positives, false positives, true negatives and false negatives statistics. But it does not actually represent a confusion matrix with the predicted vs expected values.

For that reason, I would rename the struct (and its `confusion_matrix_map` field). Since it is related to type I and type II errors, maybe the name should be related to that (e.g., `StatsError`).

It might even be that it could be simply a function to compute what's required 🤔 but for now let's start with that.

Reply:
I see your point, I called it confusion matrix because it was more intuitive of a name. I propose then ConfusionStats, I don't love Stats Error because it also includes statistics of correct values. What do you think?